File: bench/arith.py
# The Python twin of arith.sl. Python's integers are arbitrary precision, so this loop never wraps
# and never promotes -- which costs something at this size and would cost much more past 2^63.


def run():
    total = i = 0

    while i < 10000000:
        total = total + i * 2 - 1
        i = i + 1

    return total


print(run())

File: bench/arith.rb
# The Ruby twin of arith.sl. Ruby's Integer is arbitrary precision, as Python's is, so this loop
# never wraps; below 2^62 a Ruby integer is an immediate (a tagged word) and costs no allocation.

def run
  total = 0
  i = 0

  while i < 10_000_000
    total = total + i * 2 - 1
    i = i + 1
  end

  total
end

puts run
